File: LinkedList/LinkedListTest.py
from LinkedList import LinkedList

# list constructor test
arr = [0, 1, 2, 3, 4]
linkedList = LinkedList(arr)
print(linkedList)

# get value test
try:
    print(linkedList.getVal(3))
except IndexError as err:
    print(err)

# remove test
try:
    linkedList.remove(0)
    print(linkedList)
    linkedList.remove(4)
    print(linkedList)
except IndexError as err:
    print(err)

# insert to start test
linkedList.insert(0)
print(linkedList)

# insert to position test
linkedList.insert(2.5, 3)
print(linkedList)

# append test
linkedList.append(5)
print(linkedList)

# iterator test
it = linkedList.iterator()
for i in range(0, 2):
    it = it.getNext()
print(it.getValue())

# clear test
linkedList.clear()
print(linkedList)

File: LinkedList/LinkedListTest.java
package LinkedList;

import LinkedList.LinkedList.Node;

public class LinkedListTest {
    
    public static void main(String[] args) {
        // array constructor test
        Integer[] arr = {1, 2, 3, 4, 5};
        LinkedList<Integer> ll = new LinkedList<>(arr);
        System.out.println(ll.toString());

        // iterator test
        Node<Integer> it = ll.getIterator();
        it = it.getNext();
        System.out.println(it.getData());

        // get value by position test
        System.out.println("Value at 'index' 3: " + ll.getDataByPos(3));

        // insert to beginning of list test
        ll.insertAtStart(0);
        System.out.println(ll.toString());

        // append to end of list test
        ll.append(6);
        System.out.println(ll.toString());

        // insert at specific position test
        ll.insert(8, 4);
        System.out.println(ll.toString());

        // remove element test
        try {
            ll.remove(2);
            ll.remove(6);
            ll.remove(0);
            ll.remove(8);
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
        System.out.println(ll.toString());

        // clear test
        ll.clear();
        System.out.println(ll.toString());
    }

}